### PR TITLE
Feature branch initial commit

### PR DESCRIPTION
**Pull request by Tripti171992**
Please review and merge the branch feature with main.

### CODE REVIEW

File: AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/ShareSkillStepDefinitions.cs [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+                Hook.test.AddScreenCaptureFromPath(CommonMethods.SaveScreenshot(driver, "AddInvalidShareSkill"), "Exception in add invalid share skill");
+                Hook.test.Fail(ex.Message);
+                Assert.Fail();
+            }
+        }
+
+        [Then(@"Invalid skill '([^']*)' should not get added")]
+        public void ThenInvalidSkillShouldNotGetAdded(string userSkill)
+        {
+            try
```
Comment (amanbansalnz) [inline AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/ShareSkillStepDefinitions.cs]: why there need to put try and catch block ? you should not need this

Comment (amanbansalnz) [reply AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/ShareSkillStepDefinitions.cs]: in steps you should only see steps which do things, please fix this everywhere

File: AdvanceSpecFlowProject/SpecFlowProject/Steps/ShareSkillSteps/ManageListingsComponentSteps.cs [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+            }
+        }
+        public void ValidateShareSkillAdded(SkillModel skill)
+        {
+            //--------Validate askill is added--------
+            try
+            {
+                //Verify the added share skill record in the table
+                string additionStatus = ManageListingsComponentObj.GetShareSkillAddedStatus(skill);
+                Assert.AreEqual("Added", additionStatus, "Actual and expected do not match. Share skill not added!!");
```
Comment (amanbansalnz) [inline AdvanceSpecFlowProject/SpecFlowProject/Steps/ShareSkillSteps/ManageListingsComponentSteps.cs]: I would move all assertions in to another class, e.g SkillsAssertionHelper its cleaner to keep all assertions related to features in one place please fix this

File: AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+
+        [When(@"I added a language '([^']*)'")]
+        public void WhenIAddedALanguage(string userLanguage)
+        {
+            //-- Add language--
+            try
+            {
+                List<LanguageModel> languageList = JsonReader.GetData<LanguageModel>(userLanguage);
+
+                //Deleting language
```
Comment (amanbansalnz) [inline AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs]: I would move the delete of all languages and add in to a method

File: AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+        {
+            ActionOnLanguageStepsObj = new ActionOnLanguageComponentSteps();
+            ProfileLanguageOverviewStepsObj = new ProfileLanguageOverviewComponentSteps();
+        }
+
+        [When(@"I added a language '([^']*)'")]
+        public void WhenIAddedALanguage(string userLanguage)
+        {
+            //-- Add language--
+            try
```
Comment (amanbansalnz) [inline AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs]: so the try and catch not needed

File: AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+                Hook.test.AddScreenCaptureFromPath(CommonMethods.SaveScreenshot(driver, "AddLanguage"), "Exception in add language validation");
+                Hook.test.Fail(ex.Message);
+                Assert.Fail();
+            }
+        }
+
+        [When(@"I delete a language '([^']*)'")]
+        public void WhenIDeleteALanguage(string userLanguage)
+        {
+            //-- Add Skill--------------------
```
Comment (amanbansalnz) [inline AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs]: add skill? or add language

File: AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+        public void ThenALanguageShouldGetUpdated(string newLanguagePath)
+        {
+            //------Validating language updation----------
+            try
+            {
+                List<UpdateLanguageModel> languageList = JsonReader.GetData<UpdateLanguageModel>(newLanguagePath);
+
+                Hook.test.AddScreenCaptureFromPath(CommonMethods.SaveScreenshot(driver, "UpdateLanguage"), "Update language validation");
+
+                foreach (var language in languageList)
```
Comment (amanbansalnz) [inline AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs]: so all your loops should be put as part of page logic, steps should just act like delegates which get and return infomation

File: AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/RegistrationStepDefinitions.cs [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+using TechTalk.SpecFlow;
+using MarsAdvanceTaskNUnit.Steps;
+using NUnit.Framework;
+
+namespace SpecFlowProject.StepDefinitions
+{
+    [Binding]
+    public class RegistrationStepDefinitions : CommonDriver
+    {
+        RegistrationSteps RegistrationStepsObj;
```
Comment (amanbansalnz) [inline AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/RegistrationStepDefinitions.cs]: same comments apply to all step definition which i have mentioned

File: AdvanceSpecFlowProject/SpecFlowProject/Steps/HomePageSteps.cs [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+using SpecFlowProject.Pages.Components.ProfileOverview;
+using SpecFlowProject.Pages.Components.ShareSkillOverview;
+using SpecFlowProject.Pages.Components.ProfileMenuTabNavigationOverview;
+
+namespace SpecFlowProject.Steps
+{
+    public class HomePageSteps
+    {
+        HomePage HomePageComponentObj;
+        ProfileOverviewComponents ProfileOverviewComponentsObj;
```
Comment (amanbansalnz) [inline AdvanceSpecFlowProject/SpecFlowProject/Steps/HomePageSteps.cs]: you could just have one step package, for NUnit you did not have a steps but for specflow you do! you could just move all this step logic into step definition

File: AdvanceSpecFlowProject/SpecFlowProject/Pages/Components/DescriptionOverview/ActionOnDescriptionComponent.cs [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+{
+    public class ActionOnDescriptionComponent : CommonDriver
+    {
+        private IWebElement textArea;
+        private IWebElement saveButton;
+        private IWebElement messageWindow;
+        public void RenderComponents()
+        {
+            //------Render component------
+            try
```
Comment (amanbansalnz) [inline AdvanceSpecFlowProject/SpecFlowProject/Pages/Components/DescriptionOverview/ActionOnDescriptionComponent.cs]: any reason for try catch blocks here?

File: AdvanceSpecFlowProject/SpecFlowProject/Steps/HomePageSteps.cs (hunk 8 above)
Comment (amanbansalnz) [reply AdvanceSpecFlowProject/SpecFlowProject/Steps/HomePageSteps.cs]: convert Steps package into Services package and rename HomePageSteps to HomePageService

File: AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/ShareSkillStepDefinitions.cs (hunk 1 above)
Comment (Tripti171992) [reply AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/ShareSkillStepDefinitions.cs]: Issue resolved. Please review.

File: AdvanceSpecFlowProject/SpecFlowProject/Steps/ShareSkillSteps/ManageListingsComponentSteps.cs (hunk 2 above)
Comment (Tripti171992) [reply AdvanceSpecFlowProject/SpecFlowProject/Steps/ShareSkillSteps/ManageListingsComponentSteps.cs]: Issue resolved. Please review.

File: AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs (hunk 3 above)
Comment (Tripti171992) [reply AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs]: Issue resolved. Please review.

File: AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs (hunk 4 above)
Comment (Tripti171992) [reply AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs]: Issue resolved. Please review.

File: AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs (hunk 5 above)
Comment (Tripti171992) [reply AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs]: Issue resolved. Please review.

File: AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs (hunk 6 above)
Comment (Tripti171992) [reply AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/LanguageFeatureStepDefinitions.cs]: Issue resolved. Please review.

File: AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/RegistrationStepDefinitions.cs (hunk 7 above)
Comment (Tripti171992) [reply AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/RegistrationStepDefinitions.cs]: Issue resolved. Please review.

File: AdvanceSpecFlowProject/SpecFlowProject/Steps/HomePageSteps.cs (hunk 8 above)
Comment (Tripti171992) [reply AdvanceSpecFlowProject/SpecFlowProject/Steps/HomePageSteps.cs]: Issue resolved. Please review.

File: AdvanceSpecFlowProject/SpecFlowProject/Pages/Components/DescriptionOverview/ActionOnDescriptionComponent.cs (hunk 9 above)
Comment (Tripti171992) [reply AdvanceSpecFlowProject/SpecFlowProject/Pages/Components/DescriptionOverview/ActionOnDescriptionComponent.cs]: Issue resolved. Please review.

**Comment by Tripti171992** [comment]
merged with main branch

**Pull request merged**